EDU-11286: Section override (Part II - Unedited files) (#1136)
vtexdocs/dev-portal-content: docs/faststore/docs/customization/building-sections/overrides/migrating-v1-to-v2.mdx

---
title: 'Upgrading from Section Override v1 to v2'
---

Section Override v2 allows you to customize the appearance and functionality of existing components in your store. Compared to v1, v2 offers an adaptable approach. We strongly recommend updating your customization methods to use Section Override v2 in your project.

>     > ⚠️ If you choose to continue using Section Override v1, please refer to the [Overriding components (v1)](https://developers.vtex.com/docs/guides/faststore/overrides-components-and-props-v1).

Reviewer: Eu acho que deveria ter uma recomendação forte de não usar o v1.

## Main differences

> | Feature                       | Overrides v1                                           | Overrides v2                                                          |

Reviewer: ```suggestion
| Feature                       | Section Override v1                                           | Section Overrides v2                                                          |
```

| ----------------------------- | ------------------------------------------------------ | --------------------------------------------------------------------- |
| Overrides per section         | One override per section.                              | Multiple overrides per section.                                       |
| Importing Section             | String passed to `section` key.                        | Section component imported and passed to `Section` key (uppercase S). |
| Native section                | The new component overrides the native one.            | The new component is based on the native one and can coexist with it. |
| Headless CMS property schema  | No customization, limited to native schemas.           | Customizable schema allowing definition of new properties.            |
| Feature availability          | Experimental feature and may not function as expected. | Stable feature.                                                       |

Check out the main differences related to the code syntax:

> - Previously, in Section Override v1, you defined the override as an object with properties for the section name and component overrides:

Reviewer: ```suggestion
- In Section Override v1, a section is overridden by specifying the section name and component:
```

    ```js
    import { SectionOverride } from "@faststore/core";

    const SECTION = "Alert" as const;

    const override: SectionOverride = {
        section: SECTION,
        components: {
            Alert: { Component: CustomAlert },  
        },
    };

    export { override };
    ```

- Section Override v2 introduces a more component-centric approach. It directly references the section component (e.g., `AlertSection` from `@faststore/core`) using a dedicated function (`getOverriddenSection`) and builds the override object around it, emphasizing the component being overridden. This shift in v2 promotes clearer code and offers more control during the override creation process.

    ```js
    import { AlertSection, getOverriddenSection } from "@faststore/core";  

    const NewAlert = getOverriddenSection({ 
    Section: AlertSection, 
    components: {  
        Alert: { Component: CustomAlert },  
    }, 
    });  

    export default NewAlert;  
    ```

## Updating your project to Section Override v2

> Make sure your store's `@faststore/core` package is version 3.0.0 or higher to work with Section Override v2. For update instructions, refer to the [release note](https://github.com/vtex/faststore/releases/tag/v3.0.0).

Reviewer: Me parece o caso de termos um RN no Dev Portal tbm

Author: Concordo! Comecei o draft do PR [aqui](https://github.com/vtexdocs/dev-portal-content/pull/1168).

### Identify existing overrides (v1)

> Identify all overrides currently implemented using Overrides v1 (refer to the [Overrides v1](https://developers.vtex.com/docs/guides/faststore/building-sections-overriding-components-and-props) documentation). Once identified, you can keep these v1 overrides as they will continue to work in your project or create new ones using v2.

Reviewer: Em alguns lugares tá Overrides e em outros Section Override

> Identify all overrides currently implemented using Overrides v1 (refer to the [Overrides v1](https://developers.vtex.com/docs/guides/faststore/building-sections-overriding-components-and-props) documentation). Once identified, you can keep these v1 overrides as they will continue to work in your project or create new ones using v2.

Reviewer: ```suggestion
Identify all overrides currently implemented using Section Override v1 (refer to the [Section Override v1](https://developers.vtex.com/docs/guides/faststore/building-sections-overriding-components-and-props) documentation). Once identified, adjust your code to use Section Override v2.
```

Reviewer: Não acho que tem que dar muito espaço para continuar usando v1, não. Isso dificulta a manutenção, etc.
Se quiser usar, ok; mas a gente tá falando para não usar hehe

    > ⚠️ We strongly recommend using Section Override v2 instead of Section Override v1.

> If you wish to maintain existing v1 overrides alongside new v2 overrides, avoid modifying the v1-overridden components. Instead, create new overrides using v2, following the guide on [Overriding a component](https://developers.vtex.com/docs/guides/faststore/overrides-native-component).

Reviewer: ```suggestion
```

Reviewer: Pq?

Author: Aqui seria mais por não atualizar e migrar de uma vez para v2. A pessoa até pode criar e modificar esses overrides, só não queremos mais porque a intenção é deprecar essa versão e a v2 tem mais esse controle na customização. Porém, não está clara essa frase e o restante da documentação já está reforçando a recomendação de usar a v2.
